### ISSUE TITLE
GSSAPI won't install on Python 3.11 beta

### ISSUE THREAD

**Issue by ronf** [pythongssapi/python-gssapi#294]
### What went wrong?

I'm attempting to install gssapi against Python 3.11.0b3 and and getting the following errors:

```
Building wheels for collected packages: gssapi
  Building wheel for gssapi (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /opt/local/bin/python3.11 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/tmp/pip-install-sossa1e3/gssapi_c71088ddc1c44a169aa9725a44abae04/setup.py'"'"'; __file__='"'"'/private/tmp/pip-install-sossa1e3/gssapi_c71088ddc1c44a169aa9725a44abae04/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/tmp/pip-wheel-6fnsrkam
       cwd: /private/tmp/pip-install-sossa1e3/gssapi_c71088ddc1c44a169aa9725a44abae04/
  Complete output (121 lines):
  In distributed package, building from C files...
  Skipping the s4u extension because it is not supported by your GSSAPI implementation...
  Skipping the cred_store extension because it is not supported by your GSSAPI implementation...
  Skipping the rfc4178 extension because it is not supported by your GSSAPI implementation...
  Skipping the rfc5587 extension because it is not supported by your GSSAPI implementation...
  Skipping the rfc5588 extension because it is not supported by your GSSAPI implementation...
  Skipping the cred_imp_exp extension because it is not supported by your GSSAPI implementation...
  Skipping the dce extension because it is not supported by your GSSAPI implementation...
  Skipping the dce_aead extension because it is not supported by your GSSAPI implementation...
  Skipping the iov_mic extension because it is not supported by your GSSAPI implementation...
  Skipping the ggf extension because it is not supported by your GSSAPI implementation...
  Skipping the set_cred_opt extension because it is not supported by your GSSAPI implementation...
  Skipping the rfc6680 extension because it is not supported by your GSSAPI implementation...
  Skipping the rfc6680_comp_oid extension because it is not supported by your GSSAPI implementation...
  Skipping the password extension because it is not supported by your GSSAPI implementation...
  Skipping the password_add extension because it is not supported by your GSSAPI implementation...
  Skipping the krb5 extension because it is not supported by your GSSAPI implementation...
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-12.0-arm64-cpython-311
  creating build/lib.macosx-12.0-arm64-cpython-311/gssapi
  copying gssapi/sec_contexts.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi
  copying gssapi/mechs.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi
  copying gssapi/creds.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi
  copying gssapi/__init__.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi
  copying gssapi/names.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi
  copying gssapi/exceptions.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi
  copying gssapi/_win_config.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi
  copying gssapi/_utils.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi
  creating build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/named_tuples.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/__init__.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  creating build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw/_enum_extensions
  copying gssapi/raw/_enum_extensions/__init__.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw/_enum_extensions
  creating build/lib.macosx-12.0-arm64-cpython-311/gssapi/tests
  copying gssapi/tests/__init__.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/tests
  copying gssapi/tests/test_raw.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/tests
  copying gssapi/tests/test_high_level.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/tests
  copying gssapi/py.typed -> build/lib.macosx-12.0-arm64-cpython-311/gssapi
  copying gssapi/raw/creds.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/ext_rfc5588.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/ext_password.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/ext_krb5.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/ext_rfc6680.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/ext_rfc6680_comp_oid.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/ext_dce.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/mech_krb5.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/ext_rfc4178.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/chan_bindings.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/message.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/misc.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/ext_set_cred_opt.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/ext_password_add.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/exceptions.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/ext_cred_store.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/types.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/ext_cred_imp_exp.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/oids.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/sec_contexts.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/ext_iov_mic.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/ext_dce_aead.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/ext_rfc5801.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/ext_rfc5587.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/names.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/ext_s4u.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  copying gssapi/raw/ext_ggf.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
  running build_ext
  building 'gssapi.raw.misc' extension
  creating build/temp.macosx-12.0-arm64-cpython-311
  creating build/temp.macosx-12.0-arm64-cpython-311/gssapi
  creating build/temp.macosx-12.0-arm64-cpython-311/gssapi/raw
  /usr/bin/clang -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -pipe -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -I/opt/local/Library/Frameworks/Python.framework/Versions/3.11/include/python3.11 -c gssapi/raw/misc.c -o build/temp.macosx-12.0-arm64-cpython-311/gssapi/raw/misc.o -DOSX_HAS_GSS_FRAMEWORK -DHAS_GSSAPI_EXT_H
  gssapi/raw/misc.c:7790:23: error: no member named 'exc_type' in 'struct _err_stackitem'
      while ((exc_info->exc_type == NULL || exc_info->exc_type == Py_None) &&
              ~~~~~~~~  ^
  gssapi/raw/misc.c:7790:53: error: no member named 'exc_type' in 'struct _err_stackitem'
      while ((exc_info->exc_type == NULL || exc_info->exc_type == Py_None) &&
                                            ~~~~~~~~  ^
  gssapi/raw/misc.c:7804:23: error: no member named 'exc_type' in 'struct _err_stackitem'
      *type = exc_info->exc_type;
              ~~~~~~~~  ^
  gssapi/raw/misc.c:7806:21: error: no member named 'exc_traceback' in 'struct _err_stackitem'
      *tb = exc_info->exc_traceback;
            ~~~~~~~~  ^
  gssapi/raw/misc.c:7820:26: error: no member named 'exc_type' in 'struct _err_stackitem'
      tmp_type = exc_info->exc_type;
                 ~~~~~~~~  ^
  gssapi/raw/misc.c:7822:24: error: no member named 'exc_traceback' in 'struct _err_stackitem'
      tmp_tb = exc_info->exc_traceback;
               ~~~~~~~~  ^
  gssapi/raw/misc.c:7823:15: error: no member named 'exc_type' in 'struct _err_stackitem'
      exc_info->exc_type = type;
      ~~~~~~~~  ^
  gssapi/raw/misc.c:7825:15: error: no member named 'exc_traceback' in 'struct _err_stackitem'
      exc_info->exc_traceback = tb;
      ~~~~~~~~  ^
  gssapi/raw/misc.c:7882:30: error: no member named 'exc_type' in 'struct _err_stackitem'
          tmp_type = exc_info->exc_type;
                     ~~~~~~~~  ^
  gssapi/raw/misc.c:7884:28: error: no member named 'exc_traceback' in 'struct _err_stackitem'
          tmp_tb = exc_info->exc_traceback;
                   ~~~~~~~~  ^
  gssapi/raw/misc.c:7885:19: error: no member named 'exc_type' in 'struct _err_stackitem'
          exc_info->exc_type = local_type;
          ~~~~~~~~  ^
  gssapi/raw/misc.c:7887:19: error: no member named 'exc_traceback' in 'struct _err_stackitem'
          exc_info->exc_traceback = local_tb;
          ~~~~~~~~  ^
  gssapi/raw/misc.c:9092:5: error: incomplete definition of type 'struct _frame'
      __Pyx_PyFrame_SetLineNumber(py_frame, py_line);
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  gssapi/raw/misc.c:466:62: note: expanded from macro '__Pyx_PyFrame_SetLineNumber'
    #define __Pyx_PyFrame_SetLineNumber(frame, lineno)  (frame)->f_lineno = (lineno)
                                                        ~~~~~~~^
  /opt/local/Library/Frameworks/Python.framework/Versions/3.11/include/python3.11/pytypedefs.h:22:16: note: forward declaration of 'struct _frame'
  typedef struct _frame PyFrameObject;
                 ^
  13 errors generated.
  error: command '/usr/bin/clang' failed with exit code 1
  ----------------------------------------
  ERROR: Failed building wheel for gssapi
  Running setup.py clean for gssapi
Failed to build gssapi
Installing collected packages: gssapi
    Running setup.py install for gssapi ... error
    ERROR: Command errored out with exit status 1:
     command: /opt/local/bin/python3.11 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/tmp/pip-install-sossa1e3/gssapi_c71088ddc1c44a169aa9725a44abae04/setup.py'"'"'; __file__='"'"'/private/tmp/pip-install-sossa1e3/gssapi_c71088ddc1c44a169aa9725a44abae04/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/tmp/pip-record-__mzixsh/install-record.txt --single-version-externally-managed --compile --install-headers /opt/local/Library/Frameworks/Python.framework/Versions/3.11/include/python3.11/gssapi
         cwd: /private/tmp/pip-install-sossa1e3/gssapi_c71088ddc1c44a169aa9725a44abae04/
    Complete output (123 lines):
    In distributed package, building from C files...
    Skipping the s4u extension because it is not supported by your GSSAPI implementation...
    Skipping the cred_store extension because it is not supported by your GSSAPI implementation...
    Skipping the rfc4178 extension because it is not supported by your GSSAPI implementation...
    Skipping the rfc5587 extension because it is not supported by your GSSAPI implementation...
    Skipping the rfc5588 extension because it is not supported by your GSSAPI implementation...
    Skipping the cred_imp_exp extension because it is not supported by your GSSAPI implementation...
    Skipping the dce extension because it is not supported by your GSSAPI implementation...
    Skipping the dce_aead extension because it is not supported by your GSSAPI implementation...
    Skipping the iov_mic extension because it is not supported by your GSSAPI implementation...
    Skipping the ggf extension because it is not supported by your GSSAPI implementation...
    Skipping the set_cred_opt extension because it is not supported by your GSSAPI implementation...
    Skipping the rfc6680 extension because it is not supported by your GSSAPI implementation...
    Skipping the rfc6680_comp_oid extension because it is not supported by your GSSAPI implementation...
    Skipping the password extension because it is not supported by your GSSAPI implementation...
    Skipping the password_add extension because it is not supported by your GSSAPI implementation...
    Skipping the krb5 extension because it is not supported by your GSSAPI implementation...
    running install
    /opt/local/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
      warnings.warn(
    running build
    running build_py
    creating build
    creating build/lib.macosx-12.0-arm64-cpython-311
    creating build/lib.macosx-12.0-arm64-cpython-311/gssapi
    copying gssapi/sec_contexts.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi
    copying gssapi/mechs.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi
    copying gssapi/creds.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi
    copying gssapi/__init__.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi
    copying gssapi/names.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi
    copying gssapi/exceptions.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi
    copying gssapi/_win_config.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi
    copying gssapi/_utils.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi
    creating build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/named_tuples.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/__init__.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    creating build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw/_enum_extensions
    copying gssapi/raw/_enum_extensions/__init__.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw/_enum_extensions
    creating build/lib.macosx-12.0-arm64-cpython-311/gssapi/tests
    copying gssapi/tests/__init__.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/tests
    copying gssapi/tests/test_raw.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/tests
    copying gssapi/tests/test_high_level.py -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/tests
    copying gssapi/py.typed -> build/lib.macosx-12.0-arm64-cpython-311/gssapi
    copying gssapi/raw/creds.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/ext_rfc5588.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/ext_password.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/ext_krb5.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/ext_rfc6680.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/ext_rfc6680_comp_oid.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/ext_dce.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/mech_krb5.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/ext_rfc4178.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/chan_bindings.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/message.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/misc.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/ext_set_cred_opt.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/ext_password_add.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/exceptions.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/ext_cred_store.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/types.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/ext_cred_imp_exp.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/oids.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/sec_contexts.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/ext_iov_mic.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/ext_dce_aead.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/ext_rfc5801.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/ext_rfc5587.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/names.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/ext_s4u.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    copying gssapi/raw/ext_ggf.pyi -> build/lib.macosx-12.0-arm64-cpython-311/gssapi/raw
    running build_ext
    building 'gssapi.raw.misc' extension
    creating build/temp.macosx-12.0-arm64-cpython-311
    creating build/temp.macosx-12.0-arm64-cpython-311/gssapi
    creating build/temp.macosx-12.0-arm64-cpython-311/gssapi/raw
    /usr/bin/clang -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -pipe -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -I/opt/local/Library/Frameworks/Python.framework/Versions/3.11/include/python3.11 -c gssapi/raw/misc.c -o build/temp.macosx-12.0-arm64-cpython-311/gssapi/raw/misc.o -DOSX_HAS_GSS_FRAMEWORK -DHAS_GSSAPI_EXT_H
    gssapi/raw/misc.c:7790:23: error: no member named 'exc_type' in 'struct _err_stackitem'
        while ((exc_info->exc_type == NULL || exc_info->exc_type == Py_None) &&
                ~~~~~~~~  ^
    gssapi/raw/misc.c:7790:53: error: no member named 'exc_type' in 'struct _err_stackitem'
        while ((exc_info->exc_type == NULL || exc_info->exc_type == Py_None) &&
                                              ~~~~~~~~  ^
    gssapi/raw/misc.c:7804:23: error: no member named 'exc_type' in 'struct _err_stackitem'
        *type = exc_info->exc_type;
                ~~~~~~~~  ^
    gssapi/raw/misc.c:7806:21: error: no member named 'exc_traceback' in 'struct _err_stackitem'
        *tb = exc_info->exc_traceback;
              ~~~~~~~~  ^
    gssapi/raw/misc.c:7820:26: error: no member named 'exc_type' in 'struct _err_stackitem'
        tmp_type = exc_info->exc_type;
                   ~~~~~~~~  ^
    gssapi/raw/misc.c:7822:24: error: no member named 'exc_traceback' in 'struct _err_stackitem'
        tmp_tb = exc_info->exc_traceback;
                 ~~~~~~~~  ^
    gssapi/raw/misc.c:7823:15: error: no member named 'exc_type' in 'struct _err_stackitem'
        exc_info->exc_type = type;
        ~~~~~~~~  ^
    gssapi/raw/misc.c:7825:15: error: no member named 'exc_traceback' in 'struct _err_stackitem'
        exc_info->exc_traceback = tb;
        ~~~~~~~~  ^
    gssapi/raw/misc.c:7882:30: error: no member named 'exc_type' in 'struct _err_stackitem'
            tmp_type = exc_info->exc_type;
                       ~~~~~~~~  ^
    gssapi/raw/misc.c:7884:28: error: no member named 'exc_traceback' in 'struct _err_stackitem'
            tmp_tb = exc_info->exc_traceback;
                     ~~~~~~~~  ^
    gssapi/raw/misc.c:7885:19: error: no member named 'exc_type' in 'struct _err_stackitem'
            exc_info->exc_type = local_type;
            ~~~~~~~~  ^
    gssapi/raw/misc.c:7887:19: error: no member named 'exc_traceback' in 'struct _err_stackitem'
            exc_info->exc_traceback = local_tb;
            ~~~~~~~~  ^
    gssapi/raw/misc.c:9092:5: error: incomplete definition of type 'struct _frame'
        __Pyx_PyFrame_SetLineNumber(py_frame, py_line);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    gssapi/raw/misc.c:466:62: note: expanded from macro '__Pyx_PyFrame_SetLineNumber'
      #define __Pyx_PyFrame_SetLineNumber(frame, lineno)  (frame)->f_lineno = (lineno)
                                                          ~~~~~~~^
    /opt/local/Library/Frameworks/Python.framework/Versions/3.11/include/python3.11/pytypedefs.h:22:16: note: forward declaration of 'struct _frame'
    typedef struct _frame PyFrameObject;
                   ^
    13 errors generated.
    error: command '/usr/bin/clang' failed with exit code 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /opt/local/bin/python3.11 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/tmp/pip-install-sossa1e3/gssapi_c71088ddc1c44a169aa9725a44abae04/setup.py'"'"'; __file__='"'"'/private/tmp/pip-install-sossa1e3/gssapi_c71088ddc1c44a169aa9725a44abae04/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/tmp/pip-record-__mzixsh/install-record.txt --single-version-externally-managed --compile --install-headers /opt/local/Library/Frameworks/Python.framework/Versions/3.11/include/python3.11/gssapi Check the logs for full command output.
```

### How do we reproduce?

Run "pip install gssapi" on the latest Python 3.11 beta.

### Component versions (python-gssapi, Kerberos, OS / distro, etc.)

OS: macOS 12.4
Python: 3.11.0b3

clang:

Apple clang version 13.1.6 (clang-1316.0.21.2.5)
Target: arm64-apple-darwin21.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

Kerberos:

Version:     Kerberos 5 release 1.20
Vendor:      Massachusetts Institute of Technology
Prefix:      /opt/local
Exec_prefix: /opt/local

Installation works fine on the same machine with Python versions 3.10 and earlier.

**Comment by ronf** [comment]
Thanks for the quick response!

Installing cython and then installing python-gssapi from source worked nicely...

**Comment by delmoral313** [comment]
Hello guys, 

I am facing the same issue here, I followed the suggested steps and It worked nicely on Linux and macOS, but It doesn't work for Windows.  

### Environment
OS: win 10
Python: 3.11.0b3

### Kerberos
Version: Kerberos 5 release 1.20
Vendor: Massachusetts Institute of Technology
Installation was carried out using the 64-bit MSI Installer. 

### Issue 
```
❯ pip install gssapi
Collecting gssapi
  Using cached gssapi-1.7.3.tar.gz (1.3 MB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [8 lines of output]
      In distributed package, building from C files...
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\oscpache\AppData\Local\Temp\pip-install-frmqioxr\gssapi_b12f21f1098f41f693b711c1e5f06a52\setup.py", line 123, in <module>
          + ['-l%s' % os.path.splitext(lib)[0] for lib in os.listdir(_libs)]
                                                          ^^^^^^^^^^^^^^^^^
      FileNotFoundError: [WinError 3] The system cannot find the path specified: 'C:\\Program Files\\MIT\\Kerberos\\lib\\amd64'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

Hoping you have an idea of how to make it work, thanks.

**Comment by delmoral313** [comment]
I already solved the issue. The problem in my case was at MIT Kerberos installation time. 

When I installed it for the first time (using the MSI Installer) I chose the **typical installation** option which installs Kerberos but not all the development files needed by Cython, this means that inside the Kerberos folder (located at C:\Program Files\MIT\Kerberos) we can find the _bin_ folder, but not _lib_ and _include_. 

After un-installing and re-installing Kerberos again choosing the **custom installation** option, and enabling the option to include all development files, it worked nicely as in Linux and macOS.
